File: implementation/scipamato/public/persistence-jooq/src/intTest/java/ch/difty/scipamato/publ/persistence/JooqCodeClassRepoIntegrationTest.java
package ch.difty.scipamato.publ.persistence;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.jooq.JooqTest;
import org.testcontainers.junit.jupiter.Testcontainers;

import ch.difty.scipamato.publ.persistence.codeclass.JooqCodeClassRepo;

@SuppressWarnings("SpellCheckingInspection")
@JooqTest
@Testcontainers
class JooqCodeClassRepoIntegrationTest {

    @Autowired
    private JooqCodeClassRepo repo;

    @Test
    void finding_inEnglish_returnsLocalizedCodeClasses() {
        assertThat(repo.find("en"))
            .extracting("name")
            .containsExactly("Exposure Agent", "Region", "Study Population", "Health Outcome", "Study Design",
                "Species", "Duration of Exposure", "Setting");
    }

    @Test
    void finding_inFrench_returnsLocalizedCodeClasses() {
        assertThat(repo.find("fr"))
            .extracting("name")
            .containsExactly("Polluant nocif", "Région", "Population", "Effets physiologique ou nocifs", "Type d'étude",
                "Espèces investigées", "Durée de l'exposition", "Site d'exposition");
    }

    @Test
    void finding_inGerman_returnsLocalizedCodeClasses() {
        assertThat(repo.find("de"))
            .extracting("name")
            .containsExactly("Schadstoffe", "Region", "Kollektiv", "Zielgrössen", "Studientyp", "Spezies", "Zeitdauer",
                "Umgebung");
    }
}

File: implementation/scipamato/public/persistence-jooq/src/intTest/kotlin/ch/difty/scipamato/publ/persistence/JooqCodeClassRepoIntegrationTest.kt
package ch.difty.scipamato.publ.persistence

import ch.difty.scipamato.publ.persistence.codeclass.JooqCodeClassRepo
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jooq.JooqTest
import org.testcontainers.junit.jupiter.Testcontainers

@JooqTest
@Testcontainers
@Suppress("FunctionName")
internal open class JooqCodeClassRepoIntegrationTest {

    @Autowired
    private lateinit var repo: JooqCodeClassRepo

    @Test
    fun finding_inEnglish_returnsLocalizedCodeClasses() {
        assertThat(repo.find("en").map { it.name }).containsExactly(
            "Exposure Agent", "Region", "Study Population", "Health Outcome",
            "Study Design", "Species", "Duration of Exposure", "Setting"
        )
    }

    @Test
    @Suppress("SpellCheckingInspection")
    fun finding_inFrench_returnsLocalizedCodeClasses() {
        assertThat(repo.find("fr").map { it.name }).containsExactly(
            "Polluant nocif", "Région", "Population", "Effets physiologique ou nocifs",
            "Type d'étude", "Espèces investigées", "Durée de l'exposition", "Site d'exposition"
        )
    }

    @Test
    @Suppress("SpellCheckingInspection")
    fun finding_inGerman_returnsLocalizedCodeClasses() {
        assertThat(repo.find("de").map { it.name }).containsExactly(
            "Schadstoffe", "Region", "Kollektiv", "Zielgrössen", "Studientyp", "Spezies", "Zeitdauer", "Umgebung"
        )
    }
}
